src/overlays/actors/ovl_En_Nimotsu/z_en_nimotsu.c

```diff
@@ -28,25 +31,96 @@ const ActorInit En_Nimotsu_InitVars = {
+        COLSHAPE_CYLINDER,
+    },
+    {
+        ELEMTYPE_UNK4,
+        { 0x00000000, 0x00, 0x00 },
+        { 0x00000000, 0x00, 0x00 },
+        TOUCH_NONE | TOUCH_SFX_NORMAL,
+        BUMP_NONE,
+        OCELEM_ON,
+    },
     { 10, 30, 0, { 0, 0, 0 } },
 };
 
-#endif
+void EnNimotsu_UpdateCollision(EnNimotsu* this, PlayState* play) {
+    Collider_UpdateCylinder(&this->actor, &this->collider);
+    CollisionCheck_SetOC(play, &play->colChkCtx, &this->collider.base);
+    Actor_UpdateBgCheckInfo(play, &this->actor, 32.0f, 30.0f, 0.0f, 4);
+}
+
+void EnNimotsu_Init(Actor* thisx, PlayState* play) {
+    EnNimotsu* this = THIS;
+
+    Collider_InitCylinder(play, &this->collider);
+    Collider_InitAndSetCylinder(play, &this->collider, &this->actor, &sCylinderInit);
+
+    this->dustTimer = 10;
+    this->actor.gravity = -0.5f;
+
+    Actor_SetScale(&this->actor, 0.01f);
+}
+
+void EnNimotsu_Destroy(Actor* thisx, PlayState* play) {
+    EnNimotsu* this = THIS;
+
+    Collider_DestroyCylinder(play, &this->collider);
+}
+
+void EnNimotsu_Update(Actor* thisx, PlayState* play) {
+    s32 pad;
+    EnNimotsu* this = THIS;
+
+    Vec3f dustPosition;
+
+    Actor_MoveWithGravity(&this->actor);
+
+    if (!(this->dustDone & 1) && (this->actor.bgCheckFlags & 1)) {
+
+        if (DECR(this->dustTimer) == 0) {
+            this->dustDone |= 1;
+        }
+
+        if ((play->state.frames % 3) == 0) {
+            dustPosition.x = this->actor.world.pos.x + randPlusMinusPoint5Scaled(15.0f);
+            dustPosition.y = this->actor.world.pos.y;
+            dustPosition.z = this->actor.world.pos.z + randPlusMinusPoint5Scaled(15.0f);
+            Actor_SpawnFloorDustRing(play, &this->actor, &dustPosition, 20.0f, 0, 2.0f, 0, 0, 0);
+        }
+    }
 
-extern ColliderCylinderInit D_80BE1FB0;
+    EnNimotsu_UpdateCollision(this, play);
+}
 
-extern UNK_TYPE D_06013380;
+void EnNimotsu_Draw(Actor* thisx, PlayState* play) {
+    s32 pad;
+    EnNimotsu* this = THIS;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Nimotsu/func_80BE1C80.s")
+    Vec3f position;
+    Vec3f scale;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Nimotsu/EnNimotsu_Init.s")
+    OPEN_DISPS(play->state.gfxCtx);
+    func_8012C28C(play->state.gfxCtx);
+    gSPMatrix(POLY_OPA_DISP++, Matrix_NewMtx(play->state.gfxCtx), G_MTX_NOPUSH | G_MTX_LOAD | G_MTX_MODELVIEW);
+    gSPDisplayList(POLY_OPA_DISP++, &gBombShopBagDL);
+    func_8012C2DC(play->state.gfxCtx);
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Nimotsu/EnNimotsu_Destroy.s")
+    position.x = this->actor.world.pos.x + 7.0f;
+    position.y = this->actor.world.pos.y;
+    position.z = this->actor.world.pos.z + 2.0f;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Nimotsu/EnNimotsu_Update.s")
+    scale.x = 0.2f;
+    scale.y = 0.2f;
+    scale.z = 0.2f;
 
-#pragma GLOBAL_ASM("asm/non_matchings/overlays/ovl_En_Nimotsu/EnNimotsu_Draw.s")
+    func_800BC620(&position, &scale, 0xFF, play);
```

Comment:
The third parameter is an alpha value, so use dec instead
```suggestion
    func_800BC620(&position, &scale, 255, play);
```